codetoanalyze/cpp/pulse/aliasing.cpp, call_ifthenderef_true_null_bad, 0, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,when calling `ifthenderef` here,parameter `x` of ifthenderef,invalid access occurs here]
codetoanalyze/cpp/pulse/aliasing.cpp, call_ifnotthenderef_false_null_bad, 0, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,when calling `ifnotthenderef` here,parameter `x` of ifnotthenderef,invalid access occurs here]
codetoanalyze/cpp/pulse/aliasing.cpp, null_test_after_deref_latent_FN, 1, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,is assigned to the null pointer,null pointer dereference part of the trace starts here,parameter `x` of null_test_after_deref_latent_FN,invalid access occurs here]
codetoanalyze/cpp/pulse/basic_string.cpp, use_range_of_invalidated_temporary_string_bad, 2, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,when calling `setLanguage` here,assigned,in call to `std::basic_string::~basic_string()` (modelled),was invalidated by `delete`,use-after-lifetime part of the trace starts here,in call to `setLanguage`,assigned,in call to `Range::Range`,parameter `str` of Range::Range,in call to `std::basic_string::data()` (modelled),assigned,return from call to `Range::Range`,return from call to `setLanguage`,when calling `Range::operator[]` here,parameter `this` of Range::operator[],invalid access occurs here]
codetoanalyze/cpp/pulse/basics.cpp, multiple_invalidations_branch_latent, 6, USE_AFTER_DELETE_LATENT, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `ptr` of multiple_invalidations_branch_latent,was invalidated by `delete`,use-after-lifetime part of the trace starts here,parameter `ptr` of multiple_invalidations_branch_latent,invalid access occurs here]
codetoanalyze/cpp/pulse/basics.cpp, multiple_invalidations_branch_latent, 6, USE_AFTER_DELETE_LATENT, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `ptr` of multiple_invalidations_branch_latent,was invalidated by `delete`,use-after-lifetime part of the trace starts here,parameter `ptr` of multiple_invalidations_branch_latent,invalid access occurs here]
codetoanalyze/cpp/pulse/basics.cpp, multiple_invalidations_loop_latent, 5, USE_AFTER_DELETE_LATENT, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `ptr` of multiple_invalidations_loop_latent,was invalidated by `delete`,use-after-lifetime part of the trace starts here,parameter `ptr` of multiple_invalidations_loop_latent,invalid access occurs here]
codetoanalyze/cpp/pulse/basics.cpp, multiple_invalidations_loop_latent, 8, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,assigned,is assigned to the null pointer,null pointer dereference part of the trace starts here,parameter `ptr` of multiple_invalidations_loop_latent,invalid access occurs here]
codetoanalyze/cpp/pulse/basics.cpp, multiple_invalidations_loop_latent, 8, USE_AFTER_DELETE_LATENT, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `ptr` of multiple_invalidations_loop_latent,was invalidated by `delete`,use-after-lifetime part of the trace starts here,parameter `ptr` of multiple_invalidations_loop_latent,invalid access occurs here]
codetoanalyze/cpp/pulse/closures.cpp, ref_capture_destroy_invoke_bad, 6, USE_AFTER_LIFETIME, no_bucket, ERROR, [invalidation part of the trace starts here,variable `s` declared here,is the address of a stack variable `s` whose lifetime has ended,use-after-lifetime part of the trace starts here,variable `s` declared here,value captured by ref as `s`,invalid access occurs here]
codetoanalyze/cpp/pulse/closures.cpp, implicit_ref_capture_destroy_invoke_bad, 6, USE_AFTER_LIFETIME, no_bucket, ERROR, [invalidation part of the trace starts here,variable `s` declared here,is the address of a stack variable `s` whose lifetime has ended,use-after-lifetime part of the trace starts here,variable `s` declared here,value captured by ref as `s`,invalid access occurs here]
codetoanalyze/cpp/pulse/closures.cpp, reassign_lambda_capture_destroy_invoke_bad, 9, USE_AFTER_LIFETIME, no_bucket, ERROR, [invalidation part of the trace starts here,variable `s` declared here,is the address of a stack variable `s` whose lifetime has ended,use-after-lifetime part of the trace starts here,variable `s` declared here,value captured by ref as `s`,invalid access occurs here]
codetoanalyze/cpp/pulse/closures.cpp, ref_capture_return_local_lambda_bad, 7, STACK_VARIABLE_ADDRESS_ESCAPE, no_bucket, ERROR, [variable `x` declared here,value captured by ref as `x`,in call to `ref_capture_return_local_lambda_bad::lambda_closures.cpp:129:12::operator()`,value captured by ref as `x`,returned,return from call to `ref_capture_return_local_lambda_bad::lambda_closures.cpp:129:12::operator()`,returned here]
codetoanalyze/cpp/pulse/closures.cpp, call_lambda_bad, 4, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,allocated by call to `new` (modelled),assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,allocated by call to `new` (modelled),assigned,when calling `call_lambda_bad::lambda_closures.cpp:164:12::operator()` here,parameter `s` of call_lambda_bad::lambda_closures.cpp:164:12::operator(),invalid access occurs here]
codetoanalyze/cpp/pulse/closures.cpp, call_lambda_std_fun_bad, 5, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,allocated by call to `new` (modelled),assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,allocated by call to `new` (modelled),assigned,when calling `call_lambda_std_fun_bad::lambda_closures.cpp:172:7::operator()` here,parameter `s` of call_lambda_std_fun_bad::lambda_closures.cpp:172:7::operator(),invalid access occurs here]
codetoanalyze/cpp/pulse/closures.cpp, call_std_fun_constructor_bad, 2, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/closures.cpp, call_std_fun_constructor_bad, 5, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,allocated by call to `new` (modelled),assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,allocated by call to `new` (modelled),assigned,when calling `call_std_fun_constructor_bad::lambda_closures.cpp:179:32::operator()` here,parameter `s` of call_std_fun_constructor_bad::lambda_closures.cpp:179:32::operator(),invalid access occurs here]
codetoanalyze/cpp/pulse/closures.cpp, capture_by_value_bad, 7, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/closures.cpp, capture_by_ref_bad, 7, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/closures.cpp, capture_by_value_init_bad, 7, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/closures.cpp, capture_by_ref_init_bad, 7, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/closures.cpp, ref_capture_by_value_bad, 8, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/closures.cpp, ref_capture_by_ref_bad, 8, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/closures.cpp, struct_capture_by_ref_bad, 7, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/closures.cpp, struct_capture_by_val_bad, 7, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/closures.cpp, struct_capture_by_val_ok_FP, 7, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/closures.cpp, update_inside_lambda_as_argument_ok_FP, 1, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `update_inside_lambda_as_argument`,is assigned to the null pointer,assigned,returned,return from call to `update_inside_lambda_as_argument`,invalid access occurs here]
codetoanalyze/cpp/pulse/conditional_temporaries.cpp, condtemp::X::__infer_inner_destructor_~X, 1, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,is assigned to the null pointer,null pointer dereference part of the trace starts here,parameter `this` of condtemp::X::__infer_inner_destructor_~X,when calling `condtemp::X::name` here,parameter `this` of condtemp::X::name,invalid access occurs here]
codetoanalyze/cpp/pulse/conditional_temporaries.cpp, condtemp::FP_track_copy_operations_one_copy_ok, 3, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/conditional_temporaries.cpp, condtemp::FP_track_copy_operations_one_copy_ok, 17, NULLPTR_DEREFERENCE, no_bucket, ERROR, [condition is false,in call to `condtemp::X::X`,in call to `condtemp::X::copy_from`,parameter `x` of condtemp::X::copy_from,taking "then" branch,assigned,is assigned to the constant 1,assigned,return from call to `condtemp::X::copy_from`,return from call to `condtemp::X::X`,taking "then" branch,is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/conditional_temporaries.cpp, condtemp::FP_track_copy_operations_complex_ok, 16, NULLPTR_DEREFERENCE, no_bucket, ERROR, [condition is true,in call to `condtemp::copy`,in call to `condtemp::X::X`,in call to `condtemp::X::copy_from`,parameter `x` of condtemp::X::copy_from,taking "then" branch,assigned,is assigned to the constant 1,assigned,return from call to `condtemp::X::copy_from`,return from call to `condtemp::X::X`,return from call to `condtemp::copy`,taking "then" branch,is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/conditionals.cpp, add_test3_latent, 3, USE_AFTER_FREE_LATENT, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `x` of add_test3_latent,was invalidated by call to `free()`,use-after-lifetime part of the trace starts here,parameter `x` of add_test3_latent,invalid access occurs here]
codetoanalyze/cpp/pulse/conditionals.cpp, add_test5_latent, 5, USE_AFTER_FREE_LATENT, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `x` of add_test5_latent,was invalidated by call to `free()`,use-after-lifetime part of the trace starts here,parameter `x` of add_test5_latent,invalid access occurs here]
codetoanalyze/cpp/pulse/const_volatile_type.cpp, test_volatile3_bad, 3, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `std::shared_ptr<int>::shared_ptr`,is assigned to the null pointer,assigned,return from call to `std::shared_ptr<int>::shared_ptr`,in call to `std::shared_ptr<int>::shared_ptr`,parameter `__r` of std::shared_ptr<int>::shared_ptr,assigned,return from call to `std::shared_ptr<int>::shared_ptr`,in call to `std::shared_ptr<int>::operator*`,parameter `this` of std::shared_ptr<int>::operator*,returned,return from call to `std::shared_ptr<int>::operator*`,invalid access occurs here]
codetoanalyze/cpp/pulse/const_volatile_type.cpp, test_volatile4_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `std::shared_ptr<int>::shared_ptr`,is assigned to the null pointer,assigned,return from call to `std::shared_ptr<int>::shared_ptr`,in call to `std::shared_ptr<int>::operator*`,parameter `this` of std::shared_ptr<int>::operator*,returned,return from call to `std::shared_ptr<int>::operator*`,invalid access occurs here]
codetoanalyze/cpp/pulse/const_volatile_type.cpp, test_const3_bad, 3, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `std::shared_ptr<int const >::shared_ptr`,is assigned to the null pointer,assigned,return from call to `std::shared_ptr<int const >::shared_ptr`,in call to `std::shared_ptr<int const >::shared_ptr`,parameter `__r` of std::shared_ptr<int const >::shared_ptr,assigned,return from call to `std::shared_ptr<int const >::shared_ptr`,in call to `std::shared_ptr<int const >::operator*`,parameter `this` of std::shared_ptr<int const >::operator*,returned,return from call to `std::shared_ptr<int const >::operator*`,invalid access occurs here]
codetoanalyze/cpp/pulse/const_volatile_type.cpp, test_const4_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `std::shared_ptr<int const >::shared_ptr`,is assigned to the null pointer,assigned,return from call to `std::shared_ptr<int const >::shared_ptr`,in call to `std::shared_ptr<int const >::operator*`,parameter `this` of std::shared_ptr<int const >::operator*,returned,return from call to `std::shared_ptr<int const >::operator*`,invalid access occurs here]
codetoanalyze/cpp/pulse/deduplication.cpp, deduplication::SomeTemplatedClass<int>::lifetime_error_bad, 2, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `a` of deduplication::SomeTemplatedClass<int>::lifetime_error_bad,when calling `deduplication::SomeTemplatedClass<int>::templated_wrapper_delete_ok` here,parameter `a` of deduplication::SomeTemplatedClass<int>::templated_wrapper_delete_ok,was invalidated by `delete`,use-after-lifetime part of the trace starts here,parameter `a` of deduplication::SomeTemplatedClass<int>::lifetime_error_bad,when calling `deduplication::SomeTemplatedClass<int>::templated_wrapper_access_ok` here,parameter `a` of deduplication::SomeTemplatedClass<int>::templated_wrapper_access_ok,invalid access occurs here]
codetoanalyze/cpp/pulse/deduplication.cpp, deduplication::SomeTemplatedClass<int*>::lifetime_error_bad, 2, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `a` of deduplication::SomeTemplatedClass<int*>::lifetime_error_bad,when calling `deduplication::SomeTemplatedClass<int*>::templated_wrapper_delete_ok` here,parameter `a` of deduplication::SomeTemplatedClass<int*>::templated_wrapper_delete_ok,was invalidated by `delete`,use-after-lifetime part of the trace starts here,parameter `a` of deduplication::SomeTemplatedClass<int*>::lifetime_error_bad,when calling `deduplication::SomeTemplatedClass<int*>::templated_wrapper_access_ok` here,parameter `a` of deduplication::SomeTemplatedClass<int*>::templated_wrapper_access_ok,invalid access occurs here]
codetoanalyze/cpp/pulse/deduplication.cpp, deduplication::templated_function_bad<_Bool>, 3, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,allocated by call to `new` (modelled),assigned,when calling `deduplication::templated_delete_function<_Bool>` here,parameter `a` of deduplication::templated_delete_function<_Bool>,was invalidated by `delete`,use-after-lifetime part of the trace starts here,allocated by call to `new` (modelled),assigned,when calling `deduplication::templated_access_function<_Bool>` here,parameter `a` of deduplication::templated_access_function<_Bool>,invalid access occurs here]
codetoanalyze/cpp/pulse/deduplication.cpp, deduplication::templated_function_bad<int>, 3, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,allocated by call to `new` (modelled),assigned,when calling `deduplication::templated_delete_function<int>` here,parameter `a` of deduplication::templated_delete_function<int>,was invalidated by `delete`,use-after-lifetime part of the trace starts here,allocated by call to `new` (modelled),assigned,when calling `deduplication::templated_access_function<int>` here,parameter `a` of deduplication::templated_access_function<int>,invalid access occurs here]
codetoanalyze/cpp/pulse/deref_after_move_example.cpp, deref_after_move_example::Person::Person, 2, MEMORY_LEAK_CPP, no_bucket, ERROR, [allocation part of the trace starts here,allocated by `new` here,memory becomes unreachable here]
codetoanalyze/cpp/pulse/deref_after_move_example.cpp, deref_after_move_example::deref_after_move_bad, 4, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `deref_after_move_example::Person::move_age`,in call to `std::unique_ptr<int,std::default_delete<int>>::unique_ptr`,in call to `std::unique_ptr<int,std::default_delete<int>>::release`,is assigned to the null pointer,assigned,return from call to `std::unique_ptr<int,std::default_delete<int>>::release`,return from call to `std::unique_ptr<int,std::default_delete<int>>::unique_ptr`,return from call to `deref_after_move_example::Person::move_age`,when calling `deref_after_move_example::Person::access_age` here,parameter `this` of deref_after_move_example::Person::access_age,in call to `std::unique_ptr<int,std::default_delete<int>>::operator*`,parameter `this` of std::unique_ptr<int,std::default_delete<int>>::operator*,in call to `std::__compressed_pair<int*,std::default_delete<int>>::first`,parameter `this` of std::__compressed_pair<int*,std::default_delete<int>>::first,in call to `std::__compressed_pair_elem<int*,0,0>::__get`,parameter `this` of std::__compressed_pair_elem<int*,0,0>::__get,returned,return from call to `std::__compressed_pair_elem<int*,0,0>::__get`,returned,return from call to `std::__compressed_pair<int*,std::default_delete<int>>::first`,returned,return from call to `std::unique_ptr<int,std::default_delete<int>>::operator*`,invalid access occurs here]
codetoanalyze/cpp/pulse/destructor.cpp, destructor::destruct_unique_deletes_pointer_bad, 3, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,allocated by call to `new` (modelled),assigned,in call to `destructor::UniquePtr<destructor::X>::UniquePtr`,parameter `y` of destructor::UniquePtr<destructor::X>::UniquePtr,assigned,return from call to `destructor::UniquePtr<destructor::X>::UniquePtr`,when calling `destructor::UniquePtr<destructor::X>::~UniquePtr` here,parameter `this` of destructor::UniquePtr<destructor::X>::~UniquePtr,when calling `destructor::UniquePtr<destructor::X>::__infer_inner_destructor_~UniquePtr` here,parameter `this` of destructor::UniquePtr<destructor::X>::__infer_inner_destructor_~UniquePtr,was invalidated by `delete`,use-after-lifetime part of the trace starts here,allocated by call to `new` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/destructor.cpp, destructor::destruct_unique_ref_deletes_pointer_bad, 3, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,allocated by call to `new` (modelled),assigned,in call to `destructor::UniquePtr<destructor::X>::UniquePtr`,parameter `y` of destructor::UniquePtr<destructor::X>::UniquePtr,assigned,return from call to `destructor::UniquePtr<destructor::X>::UniquePtr`,when calling `destructor::UniquePtr<destructor::X>::~UniquePtr` here,parameter `this` of destructor::UniquePtr<destructor::X>::~UniquePtr,when calling `destructor::UniquePtr<destructor::X>::__infer_inner_destructor_~UniquePtr` here,parameter `this` of destructor::UniquePtr<destructor::X>::__infer_inner_destructor_~UniquePtr,was invalidated by `delete`,use-after-lifetime part of the trace starts here,allocated by call to `new` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/destructor.cpp, destructor::destruct_unique_ptr_deletes_pointer_bad, 5, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,allocated by call to `new` (modelled),assigned,in call to `destructor::UniquePtr<destructor::X>::UniquePtr`,parameter `y` of destructor::UniquePtr<destructor::X>::UniquePtr,assigned,return from call to `destructor::UniquePtr<destructor::X>::UniquePtr`,when calling `destructor::UniquePtr<destructor::X>::~UniquePtr` here,parameter `this` of destructor::UniquePtr<destructor::X>::~UniquePtr,when calling `destructor::UniquePtr<destructor::X>::__infer_inner_destructor_~UniquePtr` here,parameter `this` of destructor::UniquePtr<destructor::X>::__infer_inner_destructor_~UniquePtr,was invalidated by `delete`,use-after-lifetime part of the trace starts here,allocated by call to `new` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/exit_test.cpp, store_exit_null_bad, 0, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,when calling `store_exit` here,parameter `x` of store_exit,invalid access occurs here]
codetoanalyze/cpp/pulse/folly_DestructorGuard.cpp, UsingDelayedDestruction::double_delete_bad, 2, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `this` of UsingDelayedDestruction::double_delete_bad,was invalidated by `delete`,use-after-lifetime part of the trace starts here,parameter `this` of UsingDelayedDestruction::double_delete_bad,invalid access occurs here]
codetoanalyze/cpp/pulse/folly_DestructorGuard.cpp, UsingDelayedDestruction::latent_double_delete_ok, 1, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,is assigned to the null pointer,null pointer dereference part of the trace starts here,parameter `this` of UsingDelayedDestruction::latent_double_delete_ok,invalid access occurs here]
codetoanalyze/cpp/pulse/frontend.cpp, frontend::deref_null_namespace_alias_ptr_bad, 4, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,when calling `frontend::some::thing::bad_ptr` here,allocated by call to `new` (modelled),assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,in call to `frontend::some::thing::bad_ptr`,allocated by call to `new` (modelled),assigned,returned,return from call to `frontend::some::thing::bad_ptr`,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/frontend.cpp, frontend::conditional_expression_bad, 5, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,condition is false,is assigned to the null pointer,assigned,taking "then" branch,is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/frontend.cpp, frontend::call_Frontend_constructor_bad, 4, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `frontend::Frontend::Frontend`,parameter `n` of frontend::Frontend::Frontend,condition is false,is assigned to the constant 1,assigned,return from call to `frontend::Frontend::Frontend`,taking "then" branch,is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/frontend.cpp, frontend::call_Frontend_constructor2_bad, 5, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `frontend::Frontend::Frontend`,parameter `n` of frontend::Frontend::Frontend,condition is false,is assigned to the constant 1,assigned,assigned,return from call to `frontend::Frontend::Frontend`,taking "then" branch,is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/frontend.cpp, frontend::call_set_field_via_local_bad, 5, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `frontend::Frontend::set_field_via_local`,parameter `n` of frontend::Frontend::set_field_via_local,in call to `malloc (null case)` (modelled),is assigned to the null pointer,assigned,taking "else" branch,condition is false,is assigned to the constant 1,assigned,return from call to `frontend::Frontend::set_field_via_local`,taking "then" branch,is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/frontend.cpp, frontend::not_boolean_bad, 8, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,allocated by call to `malloc` (modelled),assigned,taking "then" branch,condition is false,is assigned to the constant 1,assigned,taking "then" branch,is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/frontend.cpp, frontend::FP_init_single_field_struct_ok, 4, NULLPTR_DEREFERENCE, no_bucket, ERROR, [variable `y` declared here,taking "then" branch,is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/funptr.cpp, test_syntactic_specialization_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `call_call_funptr[specialized with functions]`,in call to `call_funptr[specialized with functions]`,in call to `assign_nullptr`,is assigned to the null pointer,assigned,return from call to `assign_nullptr`,return from call to `call_funptr[specialized with functions]`,return from call to `call_call_funptr[specialized with functions]`,invalid access occurs here]
codetoanalyze/cpp/pulse/funptr.cpp, test_returned_funptr_specialization_bad, 3, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `call_call_funptr[specialized with functions]`,in call to `call_funptr[specialized with functions]`,in call to `assign_nullptr`,is assigned to the null pointer,assigned,return from call to `assign_nullptr`,return from call to `call_funptr[specialized with functions]`,return from call to `call_call_funptr[specialized with functions]`,invalid access occurs here]
codetoanalyze/cpp/pulse/funptr.cpp, funptr_if_bad, 10, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `assign_nullptr`,is assigned to the null pointer,assigned,return from call to `assign_nullptr`,invalid access occurs here]
codetoanalyze/cpp/pulse/funptr.cpp, funptr_else_bad, 10, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `assign_nullptr`,is assigned to the null pointer,assigned,return from call to `assign_nullptr`,invalid access occurs here]
codetoanalyze/cpp/pulse/funptr.cpp, funptr_conditional_call, 9, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [in call to `assign_nullptr`,is assigned to the null pointer,assigned,return from call to `assign_nullptr`,invalid access occurs here]
codetoanalyze/cpp/pulse/funptr.cpp, funptr_conditional_call_bad, 0, NULLPTR_DEREFERENCE, no_bucket, ERROR, [calling context starts here,in call to `funptr_conditional_call`,null pointer dereference part of the trace starts here,in call to `assign_nullptr`,is assigned to the null pointer,assigned,return from call to `assign_nullptr`,invalid access occurs here]
codetoanalyze/cpp/pulse/funptr.cpp, funptr_apply_funptr_with_intptrptr_specialized_bad, 11, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `apply_funptr_with_intptrptr[specialized with functions]`,in call to `assign_nullptr`,is assigned to the null pointer,assigned,return from call to `assign_nullptr`,return from call to `apply_funptr_with_intptrptr[specialized with functions]`,invalid access occurs here]
codetoanalyze/cpp/pulse/funptr.cpp, apply_funptr_with_intptrptr_and_after[specialized with functions], 8, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `apply_funptr_with_intptrptr[specialized with functions]`,in call to `assign_nullptr`,is assigned to the null pointer,assigned,return from call to `assign_nullptr`,return from call to `apply_funptr_with_intptrptr[specialized with functions]`,when calling `dereference_dereference_ptr` here,parameter `ptr` of dereference_dereference_ptr,invalid access occurs here]
codetoanalyze/cpp/pulse/funptr.cpp, funptr_conditionnaly_apply_funptr_with_intptrptr_unspecialized_bad, 11, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `conditionnaly_apply_funptr_with_intptrptr`,is assigned to the null pointer,assigned,return from call to `conditionnaly_apply_funptr_with_intptrptr`,invalid access occurs here]
codetoanalyze/cpp/pulse/funptr.cpp, funptr_conditionnaly_apply_funptr_with_intptrptr_specialized_bad, 11, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `conditionnaly_apply_funptr_with_intptrptr[specialized with functions]`,is assigned to the null pointer,assigned,return from call to `conditionnaly_apply_funptr_with_intptrptr[specialized with functions]`,invalid access occurs here]
codetoanalyze/cpp/pulse/funptr.cpp, test_assign_nullptr_callback_bad, 3, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `apply_callback[specialized with functions]`,in call to `FunPtrCallback::apply[specialized with functions]`,in call to `assign_nullptr`,is assigned to the null pointer,assigned,return from call to `assign_nullptr`,return from call to `FunPtrCallback::apply[specialized with functions]`,return from call to `apply_callback[specialized with functions]`,invalid access occurs here]
codetoanalyze/cpp/pulse/interprocedural.cpp, delete_then_read_bad, 2, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `x` of delete_then_read_bad,was invalidated by `delete`,use-after-lifetime part of the trace starts here,parameter `x` of delete_then_read_bad,when calling `wraps_read` here,parameter `x` of wraps_read,when calling `wraps_read_inner` here,parameter `x` of wraps_read_inner,invalid access occurs here]
codetoanalyze/cpp/pulse/interprocedural.cpp, delete_aliased_then_read_bad, 4, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `x` of delete_aliased_then_read_bad,assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,parameter `x` of delete_aliased_then_read_bad,assigned,when calling `wraps_read` here,parameter `x` of wraps_read,when calling `wraps_read_inner` here,parameter `x` of wraps_read_inner,invalid access occurs here]
codetoanalyze/cpp/pulse/interprocedural.cpp, delete_then_write_bad, 2, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `x` of delete_then_write_bad,when calling `wraps_delete` here,parameter `x` of wraps_delete,when calling `wraps_delete_inner` here,parameter `x` of wraps_delete_inner,was invalidated by `delete`,use-after-lifetime part of the trace starts here,parameter `x` of delete_then_write_bad,when calling `wraps_read` here,parameter `x` of wraps_read,when calling `wraps_read_inner` here,parameter `x` of wraps_read_inner,invalid access occurs here]
codetoanalyze/cpp/pulse/interprocedural.cpp, delete_inner_then_write_bad, 2, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `x` of delete_inner_then_write_bad,when calling `wraps_delete_inner` here,parameter `x` of wraps_delete_inner,was invalidated by `delete`,use-after-lifetime part of the trace starts here,parameter `x` of delete_inner_then_write_bad,when calling `wraps_read` here,parameter `x` of wraps_read,when calling `wraps_read_inner` here,parameter `x` of wraps_read_inner,invalid access occurs here]
codetoanalyze/cpp/pulse/interprocedural.cpp, latent_read_write_then_delete_ok, 1, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,is assigned to the null pointer,null pointer dereference part of the trace starts here,parameter `x` of latent_read_write_then_delete_ok,when calling `wraps_write` here,parameter `x` of wraps_write,when calling `wraps_write_inner` here,parameter `x` of wraps_write_inner,invalid access occurs here]
codetoanalyze/cpp/pulse/interprocedural.cpp, feed_invalid_into_access_bad, 2, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,when calling `may_return_invalid_ptr_ok` here,allocated by call to `new` (modelled),assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,in call to `may_return_invalid_ptr_ok`,allocated by call to `new` (modelled),assigned,returned,return from call to `may_return_invalid_ptr_ok`,assigned,when calling `call_store` here,parameter `y` of call_store,when calling `store` here,parameter `y` of store,invalid access occurs here]
codetoanalyze/cpp/pulse/interprocedural.cpp, access_to_invalidated_alias_bad, 3, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `x` of access_to_invalidated_alias_bad,when calling `invalidate_and_set_to_null` here,parameter `x_ptr` of invalidate_and_set_to_null,was invalidated by `delete`,use-after-lifetime part of the trace starts here,parameter `x` of access_to_invalidated_alias_bad,assigned,when calling `wraps_read` here,parameter `x` of wraps_read,when calling `wraps_read_inner` here,parameter `x` of wraps_read_inner,invalid access occurs here]
codetoanalyze/cpp/pulse/interprocedural.cpp, access_to_invalidated_alias2_bad, 3, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `x` of access_to_invalidated_alias2_bad,assigned,when calling `invalidate_and_set_to_null` here,parameter `x_ptr` of invalidate_and_set_to_null,was invalidated by `delete`,use-after-lifetime part of the trace starts here,parameter `x` of access_to_invalidated_alias2_bad,when calling `wraps_read` here,parameter `x` of wraps_read,when calling `wraps_read_inner` here,parameter `x` of wraps_read_inner,invalid access occurs here]
codetoanalyze/cpp/pulse/interprocedural.cpp, set_x_then_crash_latent, 2, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,assigned,is assigned to the null pointer,null pointer dereference part of the trace starts here,parameter `x` of set_x_then_crash_latent,when calling `set_first_non_null_ok` here,parameter `y` of set_first_non_null_ok,invalid access occurs here]
codetoanalyze/cpp/pulse/interprocedural.cpp, set_x_then_crash_latent, 4, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/join.cpp, invalidate_node_alias_latent, 12, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,assigned,is assigned to the null pointer,null pointer dereference part of the trace starts here,parameter `head` of invalidate_node_alias_latent,assigned,taking "then" branch,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/join.cpp, invalidate_node_alias_latent, 12, USE_AFTER_DELETE_LATENT, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `head` of invalidate_node_alias_latent,assigned,taking "then" branch,assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,parameter `head` of invalidate_node_alias_latent,assigned,taking "then" branch,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/join.cpp, invalidate_node_alias_latent, 12, USE_AFTER_DELETE_LATENT, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `cond` of invalidate_node_alias_latent,assigned,taking "else" branch,assigned,assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,parameter `cond` of invalidate_node_alias_latent,assigned,taking "else" branch,assigned,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/join.cpp, invalidate_node_alias_bad, 1, USE_AFTER_DELETE, no_bucket, ERROR, [calling context starts here,in call to `invalidate_node_alias_latent`,invalidation part of the trace starts here,parameter `head` of invalidate_node_alias_latent,assigned,taking "then" branch,assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,parameter `head` of invalidate_node_alias_latent,assigned,taking "then" branch,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/leaks.cpp, leaks::simple_new_bad, 0, MEMORY_LEAK_CPP, no_bucket, ERROR, [allocation part of the trace starts here,allocated by `new` here,memory becomes unreachable here]
codetoanalyze/cpp/pulse/leaks.cpp, leaks::simple_new_array_bad, 0, MEMORY_LEAK_CPP, no_bucket, ERROR, [allocation part of the trace starts here,allocated by `new[]` here,memory becomes unreachable here]
codetoanalyze/cpp/pulse/leaks.cpp, leaks::new_array_delete_bad, 2, MEMORY_LEAK_CPP, no_bucket, ERROR, [allocation part of the trace starts here,allocated by `new[]` here,memory becomes unreachable here]
codetoanalyze/cpp/pulse/leaks.cpp, leaks::unique_ptr_pointer_bad, 1, MEMORY_LEAK_CPP, no_bucket, ERROR, [allocation part of the trace starts here,allocated by `new` here,memory becomes unreachable here]
codetoanalyze/cpp/pulse/leaks.cpp, leaks::FP_call_static_local_alloc_twice_ok, 3, MEMORY_LEAK_CPP, no_bucket, ERROR, [allocation part of the trace starts here,when calling `leaks::static_local_alloc_ok` here,allocated by `new` here,memory becomes unreachable here]
codetoanalyze/cpp/pulse/leaks.cpp, leaks::unknown_wrapper_alloc_then_leak_bad, 3, MEMORY_LEAK_CPP, no_bucket, ERROR, [allocation part of the trace starts here,allocated by `new` here,memory becomes unreachable here]
codetoanalyze/cpp/pulse/nullptr.cpp, deref_nullptr_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/nullptr.cpp, no_check_return_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `may_return_nullptr`,in call to function `choice` with no summary,taking "then" branch,is assigned to the null pointer,returned,return from call to `may_return_nullptr`,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/nullptr.cpp, std_true_type_deref_bad, 3, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/nullptr.cpp, std_false_type_deref_bad, 3, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/nullptr.cpp, call_test_after_dereference_bad, 1, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,when calling `FN_test_after_dereference_latent` here,parameter `x` of FN_test_after_dereference_latent,invalid access occurs here]
codetoanalyze/cpp/pulse/nullptr.cpp, test_after_dereference2_latent, 1, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,is assigned to the null pointer,null pointer dereference part of the trace starts here,parameter `x` of test_after_dereference2_latent,invalid access occurs here]
codetoanalyze/cpp/pulse/nullptr.cpp, call_test_after_dereference2_bad, 1, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,when calling `test_after_dereference2_latent` here,parameter `x` of test_after_dereference2_latent,invalid access occurs here]
codetoanalyze/cpp/pulse/nullptr.cpp, SomeDerivedClass::SomeDerivedClass, 0, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,parameter `ptr` of SomeDerivedClass::SomeDerivedClass,when calling `SomeClass::SomeClass` here,assigned,is assigned to the null pointer,null pointer dereference part of the trace starts here,parameter `ptr` of SomeDerivedClass::SomeDerivedClass,invalid access occurs here]
codetoanalyze/cpp/pulse/nullptr.cpp, createSomeDerivedClass_from_unknown_function_ok, 1, NULLPTR_DEREFERENCE, no_bucket, ERROR, [*** SUPPRESSED ***,calling context starts here,in call to `SomeDerivedClass::SomeDerivedClass`,source of the null value part of the trace starts here,parameter `ptr` of SomeDerivedClass::SomeDerivedClass,when calling `SomeClass::SomeClass` here,assigned,is assigned to the null pointer,null pointer dereference part of the trace starts here,parameter `ptr` of SomeDerivedClass::SomeDerivedClass,invalid access occurs here]
codetoanalyze/cpp/pulse/nullptr.cpp, test_call_nullptr_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/nullptr.cpp, call_incr_deref_with_alias_bad, 7, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `incr_deref[specialized with aliases]`,parameter `x` of incr_deref[specialized with aliases],is assigned to the constant 1,assigned,is assigned to the constant 1,assigned,return from call to `incr_deref[specialized with aliases]`,taking "then" branch,is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/nullptr.cpp, test_capture_alias_bad, 11, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `test_capture_alias_bad::lambda_nullptr.cpp:319:21::operator()[specialized with aliases]`,value captured by value as `ptr`,is assigned to the constant 1,assigned,is assigned to the constant 1,assigned,return from call to `test_capture_alias_bad::lambda_nullptr.cpp:319:21::operator()[specialized with aliases]`,taking "then" branch,is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/optional.cpp, not_none_check_value_ok_FP, 5, OPTIONAL_EMPTY_ACCESS, no_bucket, ERROR, [in call to `folly::Optional::Optional(Value arg)` (modelled),assigned,taking "then" branch,in call to `folly::Optional::Optional(=None)` (modelled),is assigned an empty value,invalid access occurs here]
codetoanalyze/cpp/pulse/optional.cpp, none_no_check_bad, 2, OPTIONAL_EMPTY_ACCESS, no_bucket, ERROR, [in call to `folly::Optional::Optional(=None)` (modelled),is assigned an empty value,invalid access occurs here]
codetoanalyze/cpp/pulse/optional.cpp, none_copy_ok, 2, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/optional.cpp, none_copy_bad, 2, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/optional.cpp, none_copy_bad, 3, OPTIONAL_EMPTY_ACCESS, no_bucket, ERROR, [in call to `folly::Optional::Optional(=None)` (modelled),is assigned an empty value,in call to `folly::Optional::Optional(folly::Optional<Value> arg)` (modelled),invalid access occurs here]
codetoanalyze/cpp/pulse/optional.cpp, assign_bad, 4, PULSE_UNNECESSARY_COPY_ASSIGNMENT, no_bucket, ERROR, [copy assigned here]
codetoanalyze/cpp/pulse/optional.cpp, assign_bad, 5, OPTIONAL_EMPTY_ACCESS, no_bucket, ERROR, [in call to `folly::Optional::Optional(=None)` (modelled),is assigned an empty value,in call to `folly::Optional<int>::operator=`,parameter `other` of folly::Optional<int>::operator=,in call to `folly::Optional::assign(folly::Optional<Value> arg)` (modelled),return from call to `folly::Optional<int>::operator=`,invalid access occurs here]
codetoanalyze/cpp/pulse/optional.cpp, assign2_bad, 4, OPTIONAL_EMPTY_ACCESS, no_bucket, ERROR, [in call to `folly::Optional<int>::operator=`,in call to `folly::Optional::reset()` (modelled),is assigned an empty value,return from call to `folly::Optional<int>::operator=`,invalid access occurs here]
codetoanalyze/cpp/pulse/optional.cpp, operator_arrow_bad, 0, OPTIONAL_EMPTY_ACCESS, no_bucket, ERROR, [in call to `folly::Optional::Optional(=None)` (modelled),is assigned an empty value,when calling `emplace` here,parameter `state` of emplace,invalid access occurs here]
codetoanalyze/cpp/pulse/optional.cpp, get_pointer_no_check_none_check_bad, 3, OPTIONAL_EMPTY_ACCESS, no_bucket, ERROR, [in call to `folly::Optional::get_pointer() empty case` (modelled),is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/optional.cpp, FP_value_or_check_value_ok, 5, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `folly::Optional::Optional(Value arg)` (modelled),in call to `folly::Optional::value_or() non-empty case` (modelled),assigned,taking "then" branch,in call to `folly::Optional::Optional(=None)` (modelled),is assigned an empty value,invalid access occurs here]
codetoanalyze/cpp/pulse/optional.cpp, test_trace_ref, 4, OPTIONAL_EMPTY_ACCESS, no_bucket, ERROR, [in call to `folly::Optional<int>::operator=`,in call to `folly::Optional::reset()` (modelled),is assigned an empty value,return from call to `folly::Optional<int>::operator=`,invalid access occurs here]
codetoanalyze/cpp/pulse/pair.cpp, pair::deref_pair_null3_bad, 2, STACK_VARIABLE_ADDRESS_ESCAPE, no_bucket, ERROR, [in call to `pair::pairOfZeroNull2`,is assigned to the null pointer,assigned,variable `C++ temporary` declared here,in call to `std::make_pair<int,_std::nullptr_t>`,parameter `__t2` of std::make_pair<int,_std::nullptr_t>,in call to `std::forward<int>`,parameter `__t` of std::forward<int>,returned,return from call to `std::forward<int>`,in call to `std::forward<std::nullptr_t>`,parameter `__t` of std::forward<std::nullptr_t>,returned,return from call to `std::forward<std::nullptr_t>`,in call to `std::pair<int,int>::pair<int,_std::nullptr_t,_nullptr>`,parameter `__u2` of std::pair<int,int>::pair<int,_std::nullptr_t,_nullptr>,in call to `std::forward<int>`,parameter `__t` of std::forward<int>,returned,return from call to `std::forward<int>`,assigned,in call to `std::forward<std::nullptr_t>`,parameter `__t` of std::forward<std::nullptr_t>,returned,return from call to `std::forward<std::nullptr_t>`,assigned,return from call to `std::pair<int,int>::pair<int,_std::nullptr_t,_nullptr>`,return from call to `std::make_pair<int,_std::nullptr_t>`,in call to `std::pair<int,int*>::pair<int,_std::nullptr_t,_nullptr>`,parameter `__p` of std::pair<int,int*>::pair<int,_std::nullptr_t,_nullptr>,in call to `std::forward<int>`,parameter `__t` of std::forward<int>,returned,return from call to `std::forward<int>`,assigned,return from call to `std::pair<int,int*>::pair<int,_std::nullptr_t,_nullptr>`,return from call to `pair::pairOfZeroNull2`,returned here]
codetoanalyze/cpp/pulse/path.cpp, only_bad_on_42_latent, 2, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [in call to `may_return_null`,parameter `x` of may_return_null,taking "then" branch,is assigned to the null pointer,returned,return from call to `may_return_null`,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/path.cpp, faulty_call_bad, 0, NULLPTR_DEREFERENCE, no_bucket, ERROR, [calling context starts here,in call to `only_bad_on_42_latent`,null pointer dereference part of the trace starts here,in call to `may_return_null`,parameter `x` of may_return_null,taking "then" branch,is assigned to the null pointer,returned,return from call to `may_return_null`,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/reference_wrapper.cpp, reference_wrapper_heap_bad, 2, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,when calling `getwrapperHeap` here,in call to `WrapsB::WrapsB`,allocated by call to `new` (modelled),assigned,return from call to `WrapsB::WrapsB`,when calling `WrapsB::~WrapsB` here,parameter `this` of WrapsB::~WrapsB,when calling `WrapsB::__infer_inner_destructor_~WrapsB` here,parameter `this` of WrapsB::__infer_inner_destructor_~WrapsB,was invalidated by `delete`,use-after-lifetime part of the trace starts here,in call to `getwrapperHeap`,in call to `WrapsB::WrapsB`,allocated by call to `new` (modelled),assigned,return from call to `WrapsB::WrapsB`,in call to `ReferenceWrapperHeap::ReferenceWrapperHeap`,parameter `a` of ReferenceWrapperHeap::ReferenceWrapperHeap,in call to `WrapsB::getb`,parameter `this` of WrapsB::getb,returned,return from call to `WrapsB::getb`,assigned,return from call to `ReferenceWrapperHeap::ReferenceWrapperHeap`,return from call to `getwrapperHeap`,invalid access occurs here]
codetoanalyze/cpp/pulse/reference_wrapper.cpp, reference_wrapper_stack_bad, 2, USE_AFTER_LIFETIME, no_bucket, ERROR, [invalidation part of the trace starts here,variable `rw` declared here,when calling `getwrapperStack` here,variable `b` declared here,is the address of a stack variable `b` whose lifetime has ended,use-after-lifetime part of the trace starts here,variable `rw` declared here,in call to `getwrapperStack`,variable `b` declared here,in call to `ReferenceWrapperStack::ReferenceWrapperStack`,parameter `bref` of ReferenceWrapperStack::ReferenceWrapperStack,assigned,return from call to `ReferenceWrapperStack::ReferenceWrapperStack`,return from call to `getwrapperStack`,invalid access occurs here]
codetoanalyze/cpp/pulse/returns.cpp, returns::return_literal_stack_reference_bad, 0, STACK_VARIABLE_ADDRESS_ESCAPE, no_bucket, ERROR, [variable `C++ temporary` declared here,returned here]
codetoanalyze/cpp/pulse/returns.cpp, returns::return_variable_stack_reference1_bad, 2, STACK_VARIABLE_ADDRESS_ESCAPE, no_bucket, ERROR, [variable `C++ temporary` declared here,assigned,returned here]
codetoanalyze/cpp/pulse/returns.cpp, returns::return_variable_stack_reference2_bad, 3, STACK_VARIABLE_ADDRESS_ESCAPE, no_bucket, ERROR, [variable `C++ temporary` declared here,assigned,assigned,returned here]
codetoanalyze/cpp/pulse/returns.cpp, returns::return_stack_pointer_bad, 2, STACK_VARIABLE_ADDRESS_ESCAPE, no_bucket, ERROR, [variable `x` declared here,returned here]
codetoanalyze/cpp/pulse/shared_ptr_constructors.cpp, shared_ptr_constructors::aliasing_member_null_bad, 4, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `shared_ptr_constructors::ERROR_aliasing_construct_from_internal`,in call to `shared_ptr_constructors::internal_null_def`,is assigned to the null pointer,assigned,return from call to `shared_ptr_constructors::internal_null_def`,in call to `std::shared_ptr<shared_ptr_constructors::A>::shared_ptr<shared_ptr_constructors::B>`,parameter `__p` of std::shared_ptr<shared_ptr_constructors::A>::shared_ptr<shared_ptr_constructors::B>,assigned,return from call to `std::shared_ptr<shared_ptr_constructors::A>::shared_ptr<shared_ptr_constructors::B>`,return from call to `shared_ptr_constructors::ERROR_aliasing_construct_from_internal`,in call to `std::shared_ptr<shared_ptr_constructors::A>::operator->`,parameter `this` of std::shared_ptr<shared_ptr_constructors::A>::operator->,returned,return from call to `std::shared_ptr<shared_ptr_constructors::A>::operator->`,invalid access occurs here]
codetoanalyze/cpp/pulse/shared_ptr_deref.cpp, shared_ptr::empty_ptr_deref_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `std::shared_ptr<int>::shared_ptr`,is assigned to the null pointer,assigned,return from call to `std::shared_ptr<int>::shared_ptr`,in call to `std::shared_ptr<int>::operator*`,parameter `this` of std::shared_ptr<int>::operator*,returned,return from call to `std::shared_ptr<int>::operator*`,invalid access occurs here]
codetoanalyze/cpp/pulse/shared_ptr_deref.cpp, shared_ptr::nullptr_ptr_deref_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `std::shared_ptr<int>::shared_ptr`,is assigned to the null pointer,assigned,return from call to `std::shared_ptr<int>::shared_ptr`,in call to `std::shared_ptr<int>::operator*`,parameter `this` of std::shared_ptr<int>::operator*,returned,return from call to `std::shared_ptr<int>::operator*`,invalid access occurs here]
codetoanalyze/cpp/pulse/shared_ptr_deref.cpp, shared_ptr::empty_ptr_field_deref_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `std::shared_ptr<shared_ptr::X>::shared_ptr`,is assigned to the null pointer,assigned,return from call to `std::shared_ptr<shared_ptr::X>::shared_ptr`,in call to `std::shared_ptr<shared_ptr::X>::get`,parameter `this` of std::shared_ptr<shared_ptr::X>::get,returned,return from call to `std::shared_ptr<shared_ptr::X>::get`,invalid access occurs here]
codetoanalyze/cpp/pulse/shared_ptr_deref.cpp, shared_ptr::empty_ptr_field_deref2_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `std::shared_ptr<shared_ptr::X>::shared_ptr`,is assigned to the null pointer,assigned,return from call to `std::shared_ptr<shared_ptr::X>::shared_ptr`,in call to `std::shared_ptr<shared_ptr::X>::operator->`,parameter `this` of std::shared_ptr<shared_ptr::X>::operator->,returned,return from call to `std::shared_ptr<shared_ptr::X>::operator->`,invalid access occurs here]
codetoanalyze/cpp/pulse/shared_ptr_deref.cpp, shared_ptr::empty_ptr_method_deref_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `std::shared_ptr<shared_ptr::X>::shared_ptr`,is assigned to the null pointer,assigned,return from call to `std::shared_ptr<shared_ptr::X>::shared_ptr`,in call to `std::shared_ptr<shared_ptr::X>::operator->`,parameter `this` of std::shared_ptr<shared_ptr::X>::operator->,returned,return from call to `std::shared_ptr<shared_ptr::X>::operator->`,invalid access occurs here]
codetoanalyze/cpp/pulse/shared_ptr_deref.cpp, shared_ptr::shared_ptr_copy_null_deref_bad, 2, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/shared_ptr_deref.cpp, shared_ptr::shared_ptr_copy_null_deref_bad, 3, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `std::shared_ptr<int>::shared_ptr`,is assigned to the null pointer,assigned,return from call to `std::shared_ptr<int>::shared_ptr`,in call to `std::shared_ptr<int>::shared_ptr`,parameter `__r` of std::shared_ptr<int>::shared_ptr,assigned,return from call to `std::shared_ptr<int>::shared_ptr`,in call to `std::shared_ptr<int>::operator*`,parameter `this` of std::shared_ptr<int>::operator*,returned,return from call to `std::shared_ptr<int>::operator*`,invalid access occurs here]
codetoanalyze/cpp/pulse/std_atomics.cpp, atomic_test::pre_increment_decrement_test_bad, 6, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/std_atomics.cpp, atomic_test::load_store_possible_npe_bad, 5, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/std_atomics.cpp, atomic_test::exchange_possible_npe_bad, 5, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/std_atomics.cpp, atomic_test::FP_compare_exchange_weak_impossible_npe1_ok, 6, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/std_atomics.cpp, atomic_test::compare_exchange_weak_possible_npe1_bad, 6, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/std_atomics.cpp, atomic_test::FP_compare_exchange_weak_impossible_npe2_ok, 6, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/std_atomics.cpp, atomic_test::compare_exchange_weak_possible_npe2_bad, 6, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/std_atomics.cpp, atomic_test::FP_compare_exchange_strong_impossible_npe1_ok, 6, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/std_atomics.cpp, atomic_test::compare_exchange_strong_possible_npe1_bad, 6, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/std_atomics.cpp, atomic_test::FP_compare_exchange_strong_impossible_npe2_ok, 6, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/std_atomics.cpp, atomic_test::compare_exchange_strong_possible_npe2_bad, 6, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/swap.cpp, swap_null_bad, 4, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,in call to `std::swap<int_*>`,parameter `__x` of std::swap<int_*>,in call to `std::move<int_*&>`,parameter `__t` of std::move<int_*&>,returned,return from call to `std::move<int_*&>`,assigned,assigned,return from call to `std::swap<int_*>`,invalid access occurs here]
codetoanalyze/cpp/pulse/taint/basics.cpp, basics::Obj::endpoint, 1, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: passed as argument #1 to basics::Obj::endpoint with kind Endpoint,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #1 to basics::Obj::string_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/basics.cpp, basics::Obj::endpoint, 2, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: passed as argument #2 to basics::Obj::endpoint with kind Endpoint,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/basics.cpp, basics::sourceToSinkDirectBad, 2, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/basics.cpp, basics::returnSourceToSinkBad, 2, TAINT_ERROR, no_bucket, ERROR, [in call to `basics::returnSource`,source of the taint here: value returned from __infer_taint_source with kind SimpleSource,return from call to `basics::returnSource`,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/basics.cpp, basics::sourceThenCallSinkBad, 2, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,when calling `basics::callSink` here,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/basics.cpp, basics::propagateBad, 3, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,when calling `basics::callSink` here,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/basics.cpp, basics::object_source_sink_bad, 2, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from basics::Obj::method_source with kind SimpleSource,passed as argument #1 to basics::Obj::method_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/basics.cpp, basics::static_source_sink_bad, 2, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from basics::Obj::static_source with kind SimpleSource,passed as argument #0 to basics::Obj::static_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/basics.cpp, basics::template_source_bad, 2, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from basics::template_source<void_*> with kind SimpleSource,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/basics.cpp, basics::string_source_bad, 2, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from basics::Obj::string_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #1 to basics::Obj::string_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/basics.cpp, basics::getenv_source_bad, 2, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from getenv with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string<std::nullptr_t>` with no summary,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #1 to basics::Obj::string_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/basics.cpp, basics::via_field_bad1, 3, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from basics::template_source<std::string> with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::operator=` with no summary,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #0 to basics::template_sink<std::string> with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/basics.cpp, basics::via_field_bad2, 2, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from basics::template_source<std::string> with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::operator=` with no summary,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #0 to basics::template_sink<std::string> with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/basics.cpp, basics::via_passthrough_bad1, 4, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from basics::Obj::string_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #1 to basics::Obj::string_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/basics.cpp, basics::taint_arg_source_bad, 3, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: passed as argument #0 to basics::Obj::taint_arg_source with kind SimpleSource,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/basics.cpp, basics::via_sanitizer_ok1_FP, 3, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from basics::Obj::string_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #1 to basics::Obj::string_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/basics.cpp, basics::unsanitized_bad, 3, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from basics::Obj::string_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #1 to basics::Obj::string_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/basics.cpp, basics::unsanitized_bad, 4, STACK_VARIABLE_ADDRESS_ESCAPE, no_bucket, ERROR, [variable `C++ temporary` declared here,is the address of a C++ temporary of type `std::basic_string<char,std::char_traits<char>,std::allocator<char>>*` whose lifetime has ended,assigned,in call to `basics::Obj::sanitizer1`,parameter `input` of basics::Obj::sanitizer1,returned,return from call to `basics::Obj::sanitizer1`,assigned,returned here]
codetoanalyze/cpp/pulse/taint/basics.cpp, basics::funCall_bad1, 0, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,when calling `basics::funCall_bad2` here,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/pointers.cpp, pointers::assign_pointer_pass_to_sink_bad1, 2, TAINT_ERROR, no_bucket, ERROR, [in call to `pointers::assign_pointer_to_source`,source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::operator=` with no summary,return from call to `pointers::assign_pointer_to_source`,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/pointers.cpp, pointers::assign_pointer_pass_to_sink_bad2, 3, TAINT_ERROR, no_bucket, ERROR, [in call to `pointers::assign_pointer_to_source`,source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::operator=` with no summary,return from call to `pointers::assign_pointer_to_source`,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/pointers.cpp, pointers::assign_source_by_reference_bad1, 3, TAINT_ERROR, no_bucket, ERROR, [in call to `pointers::assign_source_by_reference`,source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::operator=` with no summary,return from call to `pointers::assign_source_by_reference`,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/pointers.cpp, pointers::assign_source_by_reference_bad2, 2, TAINT_ERROR, no_bucket, ERROR, [in call to `pointers::assign_source_by_reference`,source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::operator=` with no summary,return from call to `pointers::assign_source_by_reference`,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/pointers.cpp, pointers::assign_source_by_reference_bad3, 3, TAINT_ERROR, no_bucket, ERROR, [in call to `pointers::call_assign_source_by_reference`,in call to `pointers::assign_source_by_reference`,source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::operator=` with no summary,return from call to `pointers::assign_source_by_reference`,return from call to `pointers::call_assign_source_by_reference`,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/pointers.cpp, pointers::funptr_bad0, 2, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/pointers.cpp, pointers::funptr_helper_bad1[specialized with functions], 1, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/pointers.cpp, pointers::funptr_helper_bad2[specialized with functions], 1, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/strings.cpp, strings::sprintf1_bad, 4, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::c_str` with no summary,in call to function `sprintf` with no summary,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string<std::nullptr_t>` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/strings.cpp, strings::sprintf2_bad, 4, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::c_str` with no summary,in call to function `sprintf` with no summary,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string<std::nullptr_t>` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/strings.cpp, strings::strcpy1_bad, 4, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::c_str` with no summary,in call to function `strcpy` with no summary,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string<std::nullptr_t>` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/strings.cpp, strings::strcpy2_bad, 4, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::c_str` with no summary,in call to function `strcpy` with no summary,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string<std::nullptr_t>` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/strings.cpp, strings::strncpy_bad, 4, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::c_str` with no summary,in call to function `strncpy` with no summary,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string<std::nullptr_t>` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/strings.cpp, strings::memcpy_bad, 4, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::c_str` with no summary,in call to function `memcpy` with no summary,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string<std::nullptr_t>` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/strings.cpp, strings::memmove_bad, 4, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::c_str` with no summary,in call to function `memmove` with no summary,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string<std::nullptr_t>` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/strings.cpp, strings::memchr_bad, 3, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::c_str` with no summary,in call to function `memchr` with no summary,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string<std::nullptr_t>` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/strings.cpp, strings::constructor1_bad, 3, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/strings.cpp, strings::constructor2_bad, 3, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/strings.cpp, strings::constructor3_bad, 3, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::begin` with no summary,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string<std::__wrap_iter<char_*>,_void>` with no summary,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/strings.cpp, strings::concat1_bad, 3, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/strings.cpp, strings::concat2_bad, 4, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::operator+=` with no summary,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/strings.cpp, strings::concat3_bad, 2, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::operator+=` with no summary,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/strings.cpp, strings::append1_bad, 2, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::append` with no summary,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/strings.cpp, strings::append2_bad, 3, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/strings.cpp, strings::assign1_bad, 2, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::assign` with no summary,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/strings.cpp, strings::assign2_bad, 3, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/strings.cpp, strings::insert1_bad, 2, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::assign` with no summary,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/strings.cpp, strings::insert2_bad, 3, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/strings.cpp, strings::replace1_bad, 2, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::replace` with no summary,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/strings.cpp, strings::replace2_bad, 3, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/strings.cpp, strings::swap_bad, 4, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::swap` with no summary,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/strings.cpp, strings::format1_bad, 3, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/strings.cpp, strings::format2_bad, 3, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/strings.cpp, strings::format3_bad, 3, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `strings::format3<std::string_&>` with no summary,in call to function `strings::Formatter<std::basic_string<char,std::char_traits<char>,std::allocator<char>>&>::str` with no summary,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/strings.cpp, strings::format4_bad, 3, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `strings::format4<std::string_&>` with no summary,in call to function `strings::Formatter<std::basic_string<char,std::char_traits<char>,std::allocator<char>>&>::str` with no summary,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/strings.cpp, strings::format_varargs_bad, 3, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `strings::format3<ae492ca366f381af>` with no summary,in call to function `strings::Formatter<char[2*1] const (&),std::basic_string<char,std::char_traits<char>,std::allocator<char>>&,char[2*1] const (&)>::str` with no summary,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/structs.cpp, structs::struct_source_bad, 2, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `mystruct::mystruct` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/unknown_code.cpp, unknown_code::direct_bad, 2, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/unknown_code.cpp, unknown_code::skip_value_bad, 3, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,in call to function `skip_value` with no summary,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/unknown_code.cpp, unknown_code::skip_pointer_bad, 3, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,in call to function `skip_pointer` with no summary,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/taint/unknown_code.cpp, unknown_code::skip_indirect_bad, 3, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from __infer_taint_source with kind SimpleSource,in call to function `std::basic_string<char,std::char_traits<char>,std::allocator<char>>::basic_string` with no summary,passed as argument #0 to __infer_taint_sink with kind SimpleSink]
codetoanalyze/cpp/pulse/temporaries.cpp, temporaries::call_mk_UniquePtr_A_deref_bad, 4, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,in call to `temporaries::mk_UniquePtr_A`,allocated by call to `new` (modelled),in call to `temporaries::UniquePtr<temporaries::A>::UniquePtr`,parameter `y` of temporaries::UniquePtr<temporaries::A>::UniquePtr,assigned,return from call to `temporaries::UniquePtr<temporaries::A>::UniquePtr`,return from call to `temporaries::mk_UniquePtr_A`,when calling `temporaries::UniquePtr<temporaries::A>::~UniquePtr` here,parameter `this` of temporaries::UniquePtr<temporaries::A>::~UniquePtr,when calling `temporaries::UniquePtr<temporaries::A>::__infer_inner_destructor_~UniquePtr` here,parameter `this` of temporaries::UniquePtr<temporaries::A>::__infer_inner_destructor_~UniquePtr,was invalidated by `delete`,use-after-lifetime part of the trace starts here,in call to `temporaries::mk_UniquePtr_A`,allocated by call to `new` (modelled),in call to `temporaries::UniquePtr<temporaries::A>::UniquePtr`,parameter `y` of temporaries::UniquePtr<temporaries::A>::UniquePtr,assigned,return from call to `temporaries::UniquePtr<temporaries::A>::UniquePtr`,return from call to `temporaries::mk_UniquePtr_A`,in call to `temporaries::UniquePtr<temporaries::A>::get`,parameter `this` of temporaries::UniquePtr<temporaries::A>::get,returned,return from call to `temporaries::UniquePtr<temporaries::A>::get`,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/temporaries.cpp, temporaries::call_mk_UniquePtr_A_copy_object_ok, 1, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/temporaries.cpp, temporaries::bind_temporary_to_const_bad, 6, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,in call to `temporaries::mk_UniquePtr_A`,allocated by call to `new` (modelled),in call to `temporaries::UniquePtr<temporaries::A>::UniquePtr`,parameter `y` of temporaries::UniquePtr<temporaries::A>::UniquePtr,assigned,return from call to `temporaries::UniquePtr<temporaries::A>::UniquePtr`,return from call to `temporaries::mk_UniquePtr_A`,when calling `temporaries::UniquePtr<temporaries::A>::~UniquePtr` here,parameter `this` of temporaries::UniquePtr<temporaries::A>::~UniquePtr,when calling `temporaries::UniquePtr<temporaries::A>::__infer_inner_destructor_~UniquePtr` here,parameter `this` of temporaries::UniquePtr<temporaries::A>::__infer_inner_destructor_~UniquePtr,was invalidated by `delete`,use-after-lifetime part of the trace starts here,in call to `temporaries::mk_UniquePtr_A`,allocated by call to `new` (modelled),in call to `temporaries::UniquePtr<temporaries::A>::UniquePtr`,parameter `y` of temporaries::UniquePtr<temporaries::A>::UniquePtr,assigned,return from call to `temporaries::UniquePtr<temporaries::A>::UniquePtr`,return from call to `temporaries::mk_UniquePtr_A`,in call to `temporaries::UniquePtr<temporaries::A>::get`,parameter `this` of temporaries::UniquePtr<temporaries::A>::get,returned,return from call to `temporaries::UniquePtr<temporaries::A>::get`,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/trace.cpp, trace_free_bad, 4, USE_AFTER_FREE, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `x` of trace_free_bad,in call to `make_alias`,parameter `src` of make_alias,assigned,return from call to `make_alias`,when calling `do_free` here,parameter `x` of do_free,assigned,was invalidated by call to `free()`,use-after-lifetime part of the trace starts here,parameter `x` of trace_free_bad,invalid access occurs here]
codetoanalyze/cpp/pulse/traces.cpp, access_destructed_string, 6, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,assigned,in call to `std::basic_string::~basic_string()` (modelled),was invalidated by `delete`,use-after-lifetime part of the trace starts here,assigned,in call to `std::basic_string::data()` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/uninit.cpp, Uninit::call_init_by_store_ok, 3, PULSE_UNINITIALIZED_VALUE, no_bucket, ERROR, [struct field address `i` created,read to uninitialized value occurs here]
codetoanalyze/cpp/pulse/uninit.cpp, Uninit::call_get_MyClass_feasible_default_bad, 1, PULSE_UNINITIALIZED_VALUE, no_bucket, ERROR, [struct field address `i` created,read to uninitialized value occurs here]
codetoanalyze/cpp/pulse/uninit.cpp, Uninit2::read_f1_bad, 2, PULSE_UNINITIALIZED_VALUE, no_bucket, ERROR, [struct field address `f1` created,when calling `Uninit2::may_read_f1` here,parameter `this` of Uninit2::may_read_f1,read to uninitialized value occurs here]
codetoanalyze/cpp/pulse/uninit.cpp, Uninit2::read_f2_bad, 2, PULSE_UNINITIALIZED_VALUE, no_bucket, ERROR, [struct field address `f1` created,when calling `Uninit2::may_read_f2` here,parameter `this` of Uninit2::may_read_f2,read to uninitialized value occurs here]
codetoanalyze/cpp/pulse/unique_ptr_deref.cpp, unique_ptr::FP_unique_ptr_create_use_ok, 1, MEMORY_LEAK_CPP, no_bucket, ERROR, [allocation part of the trace starts here,when calling `unique_ptr::X::X` here,allocated by `new` here,memory becomes unreachable here]
codetoanalyze/cpp/pulse/unique_ptr_deref.cpp, unique_ptr::unique_ptr_release_bad, 1, MEMORY_LEAK_CPP, no_bucket, ERROR, [allocation part of the trace starts here,when calling `unique_ptr::X::X` here,allocated by `new` here,memory becomes unreachable here]
codetoanalyze/cpp/pulse/unique_ptr_deref.cpp, unique_ptr::reset_ptr_null_deref_bad, 3, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,in call to `std::unique_ptr<int,std::default_delete<int>>::reset`,parameter `__p` of std::unique_ptr<int,std::default_delete<int>>::reset,assigned,return from call to `std::unique_ptr<int,std::default_delete<int>>::reset`,in call to `std::unique_ptr<int,std::default_delete<int>>::operator*`,parameter `this` of std::unique_ptr<int,std::default_delete<int>>::operator*,in call to `std::__compressed_pair<int*,std::default_delete<int>>::first`,parameter `this` of std::__compressed_pair<int*,std::default_delete<int>>::first,in call to `std::__compressed_pair_elem<int*,0,0>::__get`,parameter `this` of std::__compressed_pair_elem<int*,0,0>::__get,returned,return from call to `std::__compressed_pair_elem<int*,0,0>::__get`,returned,return from call to `std::__compressed_pair<int*,std::default_delete<int>>::first`,returned,return from call to `std::unique_ptr<int,std::default_delete<int>>::operator*`,invalid access occurs here]
codetoanalyze/cpp/pulse/unique_ptr_deref.cpp, unique_ptr::reset_ptr_null_deref2_bad, 4, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,in call to `std::unique_ptr<int,std::default_delete<int>>::reset`,parameter `__p` of std::unique_ptr<int,std::default_delete<int>>::reset,assigned,return from call to `std::unique_ptr<int,std::default_delete<int>>::reset`,in call to `std::unique_ptr<int,std::default_delete<int>>::operator*`,parameter `this` of std::unique_ptr<int,std::default_delete<int>>::operator*,in call to `std::__compressed_pair<int*,std::default_delete<int>>::first`,parameter `this` of std::__compressed_pair<int*,std::default_delete<int>>::first,in call to `std::__compressed_pair_elem<int*,0,0>::__get`,parameter `this` of std::__compressed_pair_elem<int*,0,0>::__get,returned,return from call to `std::__compressed_pair_elem<int*,0,0>::__get`,returned,return from call to `std::__compressed_pair<int*,std::default_delete<int>>::first`,returned,return from call to `std::unique_ptr<int,std::default_delete<int>>::operator*`,invalid access occurs here]
codetoanalyze/cpp/pulse/unique_ptr_deref.cpp, unique_ptr::unique_ptr_assign_null_deref_bad, 1, MEMORY_LEAK_CPP, no_bucket, ERROR, [allocation part of the trace starts here,allocated by `new` here,memory becomes unreachable here]
codetoanalyze/cpp/pulse/unique_ptr_deref.cpp, unique_ptr::FP_unique_ptr_move_deref_ok, 1, MEMORY_LEAK_CPP, no_bucket, ERROR, [allocation part of the trace starts here,allocated by `new` here,memory becomes unreachable here]
codetoanalyze/cpp/pulse/unique_ptr_deref.cpp, unique_ptr::FP_unique_ptr_assign_deref_ok, 1, MEMORY_LEAK_CPP, no_bucket, ERROR, [allocation part of the trace starts here,allocated by `new` here,memory becomes unreachable here]
codetoanalyze/cpp/pulse/unique_ptr_deref.cpp, unique_ptr::unique_ptr_move_null_deref_bad, 3, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to `std::unique_ptr<int,std::default_delete<int>>::unique_ptr`,in call to `std::unique_ptr<int,std::default_delete<int>>::release`,is assigned to the null pointer,assigned,return from call to `std::unique_ptr<int,std::default_delete<int>>::release`,return from call to `std::unique_ptr<int,std::default_delete<int>>::unique_ptr`,in call to `std::unique_ptr<int,std::default_delete<int>>::operator*`,parameter `this` of std::unique_ptr<int,std::default_delete<int>>::operator*,in call to `std::__compressed_pair<int*,std::default_delete<int>>::first`,parameter `this` of std::__compressed_pair<int*,std::default_delete<int>>::first,in call to `std::__compressed_pair_elem<int*,0,0>::__get`,parameter `this` of std::__compressed_pair_elem<int*,0,0>::__get,returned,return from call to `std::__compressed_pair_elem<int*,0,0>::__get`,returned,return from call to `std::__compressed_pair<int*,std::default_delete<int>>::first`,returned,return from call to `std::unique_ptr<int,std::default_delete<int>>::operator*`,invalid access occurs here]
codetoanalyze/cpp/pulse/unique_ptr_deref.cpp, unique_ptr::no_delete_in_leaky_unique_ptr_bad, 0, MEMORY_LEAK_CPP, no_bucket, ERROR, [allocation part of the trace starts here,when calling `unique_ptr::X::X` here,allocated by `new` here,memory becomes unreachable here]
codetoanalyze/cpp/pulse/unknown_functions.cpp, const_no_init_bad, 3, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/unknown_functions.cpp, call_init_with_pointer_value_bad, 3, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, copy_decl_bad, 1, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, source_mod_param_ok_FP, 1, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, copy_in_branch_bad, 3, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, multiple_copies_bad, 2, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, multiple_copies_bad, 4, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, copy_via_constructor_read_bad, 2, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, copy_vec_bad, 0, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, copy_own_vec_bad, 2, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, loop_via_copy_bad, 1, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, copy_in_both_cases_bad, 3, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, copy_in_both_cases_branch_bad, 3, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, copy_modified_after_abort_ok_FP, 1, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, copy_modified_after_abort_ok_FP, 5, VECTOR_INVALIDATION, no_bucket, ERROR, [invalidation part of the trace starts here,variable `vec` declared here,was potentially invalidated by `std::vector::push_back()`,use-after-lifetime part of the trace starts here,variable `vec` declared here,in call to `std::vector::at()` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, copy_via_model_bad, 1, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, check_before_custom_destructor_bad, 0, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, copy_vec_name_contains_copy_ok, 1, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [*** SUPPRESSED ***,copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, source_const_fcn_bad, 1, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, source_dispatch_const_fcn_bad, 1, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, constructor_bad, 2, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, copy_assignment_bad, 2, PULSE_UNNECESSARY_COPY_ASSIGNMENT, no_bucket, ERROR, [copy assigned here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, copy_and_move_bad, 1, PULSE_UNNECESSARY_COPY, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, WrapperArr::WrapperArr, 0, PULSE_UNNECESSARY_COPY_MOVABLE, no_bucket, ERROR, [copied here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, WrapperArr::unnecessary_copy_moveable_bad, 1, PULSE_UNNECESSARY_COPY_ASSIGNMENT_MOVABLE, no_bucket, ERROR, [copy assigned here]
codetoanalyze/cpp/pulse/unnecessary_copy.cpp, WrapperArr::unnecessary_copy_moveable_copy_mod_bad, 1, PULSE_UNNECESSARY_COPY_ASSIGNMENT_MOVABLE, no_bucket, ERROR, [copy assigned here]
codetoanalyze/cpp/pulse/use_after_delete.cpp, deref_deleted_bad, 3, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,allocated by call to `new` (modelled),assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,allocated by call to `new` (modelled),assigned,when calling `Simple::Simple` here,parameter `__param_0` of Simple::Simple,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_delete.cpp, reassign_field_of_deleted_bad, 3, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,allocated by call to `new` (modelled),assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,allocated by call to `new` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_delete.cpp, double_delete_bad, 3, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,allocated by call to `new` (modelled),assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,allocated by call to `new` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_delete.cpp, delete_in_branch_latent, 5, USE_AFTER_DELETE_LATENT, no_bucket, ERROR, [invalidation part of the trace starts here,allocated by call to `new` (modelled),assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,allocated by call to `new` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_delete.cpp, use_in_branch_latent, 4, USE_AFTER_DELETE_LATENT, no_bucket, ERROR, [invalidation part of the trace starts here,allocated by call to `new` (modelled),assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,allocated by call to `new` (modelled),assigned,when calling `Simple::Simple` here,parameter `__param_0` of Simple::Simple,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_delete.cpp, delete_in_loop_bad, 3, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,allocated by call to `new` (modelled),assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,allocated by call to `new` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_delete.cpp, use_in_loop_bad, 4, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,allocated by call to `new` (modelled),assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,allocated by call to `new` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_delete.cpp, delete_allocated_then_error_bad, 1, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [*** SUPPRESSED ***,source of the null value part of the trace starts here,is assigned to the null pointer,null pointer dereference part of the trace starts here,parameter `x` of delete_allocated_then_error_bad,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_delete.cpp, delete_allocated_then_error_bad, 3, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `x` of delete_allocated_then_error_bad,was invalidated by `delete`,use-after-lifetime part of the trace starts here,parameter `x` of delete_allocated_then_error_bad,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_delete.cpp, null_call_delete_allocated_then_error_bad, 1, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,when calling `delete_allocated_then_error_bad` here,parameter `x` of delete_allocated_then_error_bad,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_destructor.cpp, use_after_destructor::reinit_after_explicit_destructor_bad, 3, PULSE_UNNECESSARY_COPY_ASSIGNMENT, no_bucket, ERROR, [copy assigned here]
codetoanalyze/cpp/pulse/use_after_destructor.cpp, use_after_destructor::reinit_after_explicit_destructor_bad, 4, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,in call to `use_after_destructor::S::S`,allocated by call to `new` (modelled),assigned,return from call to `use_after_destructor::S::S`,when calling `use_after_destructor::S::~S` here,parameter `this` of use_after_destructor::S::~S,when calling `use_after_destructor::S::__infer_inner_destructor_~S` here,parameter `this` of use_after_destructor::S::__infer_inner_destructor_~S,was invalidated by `delete`,use-after-lifetime part of the trace starts here,in call to `use_after_destructor::S::S`,allocated by call to `new` (modelled),assigned,return from call to `use_after_destructor::S::S`,in call to `use_after_destructor::S::operator=`,parameter `__param_0` of use_after_destructor::S::operator=,assigned,return from call to `use_after_destructor::S::operator=`,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_destructor.cpp, use_after_destructor::reinit_after_explicit_destructor2_bad, 4, PULSE_UNNECESSARY_COPY_ASSIGNMENT, no_bucket, ERROR, [copy assigned here]
codetoanalyze/cpp/pulse/use_after_destructor.cpp, use_after_destructor::reinit_after_explicit_destructor2_bad, 5, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,in call to `use_after_destructor::S::S`,allocated by call to `new` (modelled),assigned,return from call to `use_after_destructor::S::S`,when calling `use_after_destructor::S::~S` here,parameter `this` of use_after_destructor::S::~S,when calling `use_after_destructor::S::__infer_inner_destructor_~S` here,parameter `this` of use_after_destructor::S::__infer_inner_destructor_~S,was invalidated by `delete`,use-after-lifetime part of the trace starts here,in call to `use_after_destructor::S::S`,allocated by call to `new` (modelled),assigned,return from call to `use_after_destructor::S::S`,in call to `use_after_destructor::S::operator=`,parameter `__param_0` of use_after_destructor::S::operator=,assigned,return from call to `use_after_destructor::S::operator=`,when calling `use_after_destructor::S::~S` here,parameter `this` of use_after_destructor::S::~S,when calling `use_after_destructor::S::__infer_inner_destructor_~S` here,parameter `this` of use_after_destructor::S::__infer_inner_destructor_~S,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_destructor.cpp, use_after_destructor::double_destructor_bad, 5, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,in call to `use_after_destructor::S::S`,allocated by call to `new` (modelled),assigned,return from call to `use_after_destructor::S::S`,when calling `use_after_destructor::S::~S` here,parameter `this` of use_after_destructor::S::~S,when calling `use_after_destructor::S::__infer_inner_destructor_~S` here,parameter `this` of use_after_destructor::S::__infer_inner_destructor_~S,was invalidated by `delete`,use-after-lifetime part of the trace starts here,in call to `use_after_destructor::S::S`,allocated by call to `new` (modelled),assigned,return from call to `use_after_destructor::S::S`,when calling `use_after_destructor::S::~S` here,parameter `this` of use_after_destructor::S::~S,when calling `use_after_destructor::S::__infer_inner_destructor_~S` here,parameter `this` of use_after_destructor::S::__infer_inner_destructor_~S,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_destructor.cpp, use_after_destructor::use_after_destructor_bad, 3, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,in call to `use_after_destructor::S::S`,allocated by call to `new` (modelled),assigned,return from call to `use_after_destructor::S::S`,when calling `use_after_destructor::S::~S` here,parameter `this` of use_after_destructor::S::~S,when calling `use_after_destructor::S::__infer_inner_destructor_~S` here,parameter `this` of use_after_destructor::S::__infer_inner_destructor_~S,was invalidated by `delete`,use-after-lifetime part of the trace starts here,in call to `use_after_destructor::S::S`,allocated by call to `new` (modelled),assigned,return from call to `use_after_destructor::S::S`,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_destructor.cpp, use_after_destructor::use_after_scope1_bad, 4, PULSE_UNNECESSARY_COPY_ASSIGNMENT, no_bucket, ERROR, [copy assigned here]
codetoanalyze/cpp/pulse/use_after_destructor.cpp, use_after_destructor::use_after_scope1_bad, 7, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,in call to `use_after_destructor::S::S`,allocated by call to `new` (modelled),assigned,return from call to `use_after_destructor::S::S`,when calling `use_after_destructor::S::~S` here,parameter `this` of use_after_destructor::S::~S,when calling `use_after_destructor::S::__infer_inner_destructor_~S` here,parameter `this` of use_after_destructor::S::__infer_inner_destructor_~S,was invalidated by `delete`,use-after-lifetime part of the trace starts here,in call to `use_after_destructor::S::S`,allocated by call to `new` (modelled),assigned,return from call to `use_after_destructor::S::S`,in call to `use_after_destructor::S::operator=`,parameter `__param_0` of use_after_destructor::S::operator=,assigned,return from call to `use_after_destructor::S::operator=`,when calling `use_after_destructor::S::~S` here,parameter `this` of use_after_destructor::S::~S,when calling `use_after_destructor::S::__infer_inner_destructor_~S` here,parameter `this` of use_after_destructor::S::__infer_inner_destructor_~S,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_destructor.cpp, use_after_destructor::use_after_scope2_bad, 2, PULSE_UNNECESSARY_COPY_ASSIGNMENT, no_bucket, ERROR, [copy assigned here]
codetoanalyze/cpp/pulse/use_after_destructor.cpp, use_after_destructor::use_after_scope2_bad, 3, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,in call to `use_after_destructor::S::S`,allocated by call to `new` (modelled),assigned,return from call to `use_after_destructor::S::S`,when calling `use_after_destructor::S::~S` here,parameter `this` of use_after_destructor::S::~S,when calling `use_after_destructor::S::__infer_inner_destructor_~S` here,parameter `this` of use_after_destructor::S::__infer_inner_destructor_~S,was invalidated by `delete`,use-after-lifetime part of the trace starts here,in call to `use_after_destructor::S::S`,allocated by call to `new` (modelled),assigned,return from call to `use_after_destructor::S::S`,in call to `use_after_destructor::S::operator=`,parameter `__param_0` of use_after_destructor::S::operator=,assigned,return from call to `use_after_destructor::S::operator=`,when calling `use_after_destructor::S::~S` here,parameter `this` of use_after_destructor::S::~S,when calling `use_after_destructor::S::__infer_inner_destructor_~S` here,parameter `this` of use_after_destructor::S::__infer_inner_destructor_~S,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_destructor.cpp, use_after_destructor::placement_new_wrong_order_bad, 4, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,in call to `use_after_destructor::S::S`,allocated by call to `new` (modelled),assigned,return from call to `use_after_destructor::S::S`,when calling `use_after_destructor::S::~S` here,parameter `this` of use_after_destructor::S::~S,when calling `use_after_destructor::S::__infer_inner_destructor_~S` here,parameter `this` of use_after_destructor::S::__infer_inner_destructor_~S,was invalidated by `delete`,use-after-lifetime part of the trace starts here,in call to `use_after_destructor::S::S`,allocated by call to `new` (modelled),assigned,return from call to `use_after_destructor::S::S`,when calling `use_after_destructor::S::~S` here,parameter `this` of use_after_destructor::S::~S,when calling `use_after_destructor::S::__infer_inner_destructor_~S` here,parameter `this` of use_after_destructor::S::__infer_inner_destructor_~S,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_destructor.cpp, use_after_destructor::placement_new_aliasing1_bad, 5, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,allocated by call to `new` (modelled),assigned,in call to `<placement new>()` (modelled),assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,allocated by call to `new` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_destructor.cpp, use_after_destructor::placement_new_aliasing2_bad, 5, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,allocated by call to `new` (modelled),assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,allocated by call to `new` (modelled),assigned,in call to `<placement new>()` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_destructor.cpp, use_after_destructor::placement_new_aliasing3_bad, 6, USE_AFTER_DELETE, no_bucket, ERROR, [invalidation part of the trace starts here,allocated by call to `new` (modelled),assigned,was invalidated by `delete`,use-after-lifetime part of the trace starts here,allocated by call to `new` (modelled),assigned,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_destructor.cpp, use_after_destructor::use_after_scope4_bad, 6, USE_AFTER_LIFETIME, no_bucket, ERROR, [invalidation part of the trace starts here,variable `c` declared here,is the address of a stack variable `c` whose lifetime has ended,use-after-lifetime part of the trace starts here,variable `c` declared here,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_free.cpp, use_after_free_simple_bad, 2, USE_AFTER_FREE, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `x` of use_after_free_simple_bad,was invalidated by call to `free()`,use-after-lifetime part of the trace starts here,parameter `x` of use_after_free_simple_bad,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_free.cpp, double_free_simple_bad, 2, USE_AFTER_FREE, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `x` of double_free_simple_bad,was invalidated by call to `free()`,use-after-lifetime part of the trace starts here,parameter `x` of double_free_simple_bad,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_free.cpp, double_free_global_bad, 2, USE_AFTER_FREE, no_bucket, ERROR, [invalidation part of the trace starts here,global variable `global_pointer` accessed here,when calling `free_global_pointer_ok` here,global variable `global_pointer` accessed here,was invalidated by call to `free()`,use-after-lifetime part of the trace starts here,global variable `global_pointer` accessed here,when calling `free_global_pointer_ok` here,global variable `global_pointer` accessed here,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_free.cpp, free_null_then_deref_bad, 3, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_free.cpp, assumed_aliasing_latent, 3, USE_AFTER_FREE_LATENT, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `x` of assumed_aliasing_latent,was invalidated by call to `free()`,use-after-lifetime part of the trace starts here,parameter `y` of assumed_aliasing_latent,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_free.cpp, trigger_assumed_aliasing_bad, 0, USE_AFTER_FREE, no_bucket, ERROR, [calling context starts here,in call to `assumed_aliasing_latent`,invalidation part of the trace starts here,parameter `x` of assumed_aliasing_latent,was invalidated by call to `free()`,use-after-lifetime part of the trace starts here,parameter `y` of assumed_aliasing_latent,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_free.cpp, assumed_aliasing2_latent, 4, USE_AFTER_FREE_LATENT, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `x` of assumed_aliasing2_latent,was invalidated by call to `free()`,use-after-lifetime part of the trace starts here,parameter `y` of assumed_aliasing2_latent,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_free.cpp, trigger_assumed_aliasing2_bad, 0, USE_AFTER_FREE, no_bucket, ERROR, [calling context starts here,in call to `assumed_aliasing2_latent`,invalidation part of the trace starts here,parameter `x` of assumed_aliasing2_latent,was invalidated by call to `free()`,use-after-lifetime part of the trace starts here,parameter `y` of assumed_aliasing2_latent,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_free.cpp, assumed_aliasing3_latent, 4, USE_AFTER_FREE_LATENT, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `x` of assumed_aliasing3_latent,was invalidated by call to `free()`,use-after-lifetime part of the trace starts here,parameter `y` of assumed_aliasing3_latent,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_free.cpp, trigger_assumed_aliasing3_bad, 0, USE_AFTER_FREE, no_bucket, ERROR, [calling context starts here,in call to `assumed_aliasing3_latent`,invalidation part of the trace starts here,parameter `x` of assumed_aliasing3_latent,was invalidated by call to `free()`,use-after-lifetime part of the trace starts here,parameter `y` of assumed_aliasing3_latent,invalid access occurs here]
codetoanalyze/cpp/pulse/use_after_scope.cpp, access_out_of_scope_stack_ref_bad, 3, USE_AFTER_LIFETIME, no_bucket, ERROR, [invalidation part of the trace starts here,variable `p` declared here,when calling `invalidate_local_ok` here,variable `t` declared here,is the address of a stack variable `t` whose lifetime has ended,use-after-lifetime part of the trace starts here,variable `p` declared here,in call to `invalidate_local_ok`,variable `t` declared here,assigned,return from call to `invalidate_local_ok`,invalid access occurs here]
codetoanalyze/cpp/pulse/values.cpp, error_under_true_conditionals_bad, 5, USE_AFTER_FREE, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `x` of error_under_true_conditionals_bad,was invalidated by call to `free()`,use-after-lifetime part of the trace starts here,parameter `x` of error_under_true_conditionals_bad,invalid access occurs here]
codetoanalyze/cpp/pulse/values.cpp, free_if_deref_bad, 2, USE_AFTER_FREE, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `x` of free_if_deref_bad,when calling `free_if` here,parameter `x` of free_if,was invalidated by call to `free()`,use-after-lifetime part of the trace starts here,parameter `x` of free_if_deref_bad,invalid access occurs here]
codetoanalyze/cpp/pulse/vector.cpp, deref_vector_element_after_push_back_bad, 4, VECTOR_INVALIDATION, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `vec` of deref_vector_element_after_push_back_bad,was potentially invalidated by `std::vector::push_back()`,use-after-lifetime part of the trace starts here,parameter `vec` of deref_vector_element_after_push_back_bad,in call to `std::vector::at()` (modelled),assigned,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/vector.cpp, deref_vector_pointer_element_after_push_back_bad, 4, VECTOR_INVALIDATION, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `vec` of deref_vector_pointer_element_after_push_back_bad,was potentially invalidated by `std::vector::push_back()`,use-after-lifetime part of the trace starts here,parameter `vec` of deref_vector_pointer_element_after_push_back_bad,in call to `std::vector::at()` (modelled),assigned,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/vector.cpp, deref_local_vector_element_after_push_back_bad, 4, VECTOR_INVALIDATION, no_bucket, ERROR, [invalidation part of the trace starts here,variable `C++ temporary` declared here,was potentially invalidated by `std::vector::push_back()`,use-after-lifetime part of the trace starts here,variable `C++ temporary` declared here,in call to `std::vector::at()` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/vector.cpp, deref_null_local_vector_element_bad, 2, NULLPTR_DEREFERENCE, no_bucket, ERROR, [is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/vector.cpp, FP_init_fill_then_push_back_ok, 7, VECTOR_INVALIDATION, no_bucket, ERROR, [invalidation part of the trace starts here,variable `vec` declared here,was potentially invalidated by `std::vector::push_back()`,use-after-lifetime part of the trace starts here,variable `vec` declared here,in call to `std::vector::at()` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/vector.cpp, push_back_loop_latent, 6, VECTOR_INVALIDATION_LATENT, no_bucket, ERROR, [invalidation part of the trace starts here,variable `vec` declared here,was potentially invalidated by `std::vector::push_back()`,use-after-lifetime part of the trace starts here,variable `vec` declared here,in call to `std::vector::at()` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/vector.cpp, reserve_bad, 3, VECTOR_INVALIDATION, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `vec` of reserve_bad,was potentially invalidated by `std::vector::reserve()`,use-after-lifetime part of the trace starts here,parameter `vec` of reserve_bad,in call to `std::vector::at()` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/vector.cpp, clear_bad, 3, VECTOR_INVALIDATION, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `vec` of clear_bad,was potentially invalidated by `std::vector::clear()`,use-after-lifetime part of the trace starts here,parameter `vec` of clear_bad,in call to `std::vector::at()` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/vector.cpp, assign_bad, 3, VECTOR_INVALIDATION, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `vec` of assign_bad,was potentially invalidated by `std::vector::assign()`,use-after-lifetime part of the trace starts here,parameter `vec` of assign_bad,in call to `std::vector::at()` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/vector.cpp, shrink_to_fit_bad, 3, VECTOR_INVALIDATION, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `vec` of shrink_to_fit_bad,was potentially invalidated by `std::vector::shrink_to_fit()`,use-after-lifetime part of the trace starts here,parameter `vec` of shrink_to_fit_bad,in call to `std::vector::at()` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/vector.cpp, insert_bad, 3, VECTOR_INVALIDATION, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `vec` of insert_bad,was potentially invalidated by `std::vector::insert()`,use-after-lifetime part of the trace starts here,parameter `vec` of insert_bad,in call to `std::vector::at()` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/vector.cpp, emplace_bad, 3, VECTOR_INVALIDATION, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `vec` of emplace_bad,was potentially invalidated by `std::vector::emplace()`,use-after-lifetime part of the trace starts here,parameter `vec` of emplace_bad,in call to `std::vector::at()` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/vector.cpp, emplace_back_bad, 3, VECTOR_INVALIDATION, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `vec` of emplace_back_bad,was potentially invalidated by `std::vector::emplace_back()`,use-after-lifetime part of the trace starts here,parameter `vec` of emplace_back_bad,in call to `std::vector::at()` (modelled),assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/vector_iterator.cpp, iterator_read_after_emplace_bad, 3, VECTOR_INVALIDATION, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `vec` of iterator_read_after_emplace_bad,was potentially invalidated by `std::vector::emplace()`,use-after-lifetime part of the trace starts here,variable `iter` declared here,in call to `std::vector::begin()` (modelled),invalid access occurs here]
codetoanalyze/cpp/pulse/vector_iterator.cpp, iterator_next_after_emplace_bad, 3, VECTOR_INVALIDATION, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `vec` of iterator_next_after_emplace_bad,was potentially invalidated by `std::vector::emplace()`,use-after-lifetime part of the trace starts here,variable `iter` declared here,in call to `std::vector::begin()` (modelled),invalid access occurs here]
codetoanalyze/cpp/pulse/vector_iterator.cpp, iterator_after_push_back_loop_latent, 9, VECTOR_INVALIDATION_LATENT, no_bucket, ERROR, [invalidation part of the trace starts here,variable `vec` declared here,was potentially invalidated by `std::vector::push_back()`,use-after-lifetime part of the trace starts here,variable `iter_begin` declared here,in call to `std::vector::begin()` (modelled),in call to `iterator constructor` (modelled),invalid access occurs here]
codetoanalyze/cpp/pulse/vector_iterator.cpp, iterator_end_read_bad, 3, VECTOR_INVALIDATION, no_bucket, ERROR, [invalidation part of the trace starts here,is pointed to by the `end()` iterator,use-after-lifetime part of the trace starts here,variable `iter` declared here,in call to `std::vector::end()` (modelled),invalid access occurs here]
codetoanalyze/cpp/pulse/vector_iterator.cpp, iterator_end_next_bad, 3, VECTOR_INVALIDATION, no_bucket, ERROR, [invalidation part of the trace starts here,is pointed to by the `end()` iterator,use-after-lifetime part of the trace starts here,variable `iter` declared here,in call to `std::vector::end()` (modelled),invalid access occurs here]
codetoanalyze/cpp/pulse/vector_iterator.cpp, iterator_prev_after_emplace_bad, 4, VECTOR_INVALIDATION, no_bucket, ERROR, [invalidation part of the trace starts here,parameter `vec` of iterator_prev_after_emplace_bad,was potentially invalidated by `std::vector::emplace()`,use-after-lifetime part of the trace starts here,variable `iter` declared here,in call to `std::vector::begin()` (modelled),invalid access occurs here]
codetoanalyze/cpp/pulse/weak_ptr.cpp, weak_ptr_constructors::fromWeakBaseAssign, 2, PULSE_UNNECESSARY_COPY_ASSIGNMENT, no_bucket, ERROR, [copy assigned here]
codetoanalyze/cpp/pulse/weak_ptr.cpp, weak_ptr_observers::use_count_empty_bad, 4, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to function `std::weak_ptr<int>::use_count` with no summary,taking "then" branch,is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/weak_ptr.cpp, weak_ptr_observers::FN_use_count_after_reset_bad, 4, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [in call to function `std::weak_ptr<int>::use_count` with no summary,taking "then" branch,is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/weak_ptr.cpp, weak_ptr_observers::expired_empty_bad, 4, NULLPTR_DEREFERENCE, no_bucket, ERROR, [in call to function `std::weak_ptr<int>::expired` with no summary,taking "then" branch,is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/weak_ptr.cpp, weak_ptr_observers::FN_expired_after_reset_bad, 4, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [in call to function `std::weak_ptr<int>::expired` with no summary,taking "then" branch,is assigned to the null pointer,assigned,invalid access occurs here]
codetoanalyze/cpp/pulse/weak_ptr.cpp, weak_ptr_observers::FN_expired_after_swap_bad, 5, NULLPTR_DEREFERENCE_LATENT, no_bucket, ERROR, [in call to function `std::weak_ptr<int>::expired` with no summary,taking "then" branch,is assigned to the null pointer,assigned,invalid access occurs here]
